src/prefect/flows.py

```diff
@@ -623,6 +623,12 @@ def my_other_flow(name):
         """
         from prefect.deployments.runner import RunnerDeployment
 
+        if "/" in name and not name.endswith(".py"):
+            raise ValueError(
+                f"Invalid deployment name {name!r}. Deployment names cannot contain"
+                " forward slashes."
+            )
+
```

Comment:
Okay I'm up to speed on why we'd put the validation logic here.. 

But we may as well cover all our bases for deployment name validation while we're at it.
```suggestion
        if not name.endswith(".py"):
            raise_on_name_with_banned_characters(name)
```

Reply:
Ah! I didn't realize that existed, but it makes sense that it does. Nice